scaladoc-js/src/Main.scala

```diff
@@ -1,9 +1,22 @@
 package dotty.tools.scaladoc
 
-object Main extends App {
-  Searchbar()
-  SocialLinks()
-  CodeSnippets()
-  DropdownHandler()
-  Ux()
-}
+object Main:
+
+  private def common(): Unit =
+    CodeSnippets()
+
+  def main(): Unit =
+    Searchbar()
+    SocialLinks()
+    DropdownHandler()
+    Ux()
+    common()
+
+  /**
+   * This main is conditionally enabled by system env variable `scaladoc.projectFormat=md`
+   * passed in ./projects/scripts/genDocsScalaLang
+   * The reason why we have to pass the condition by env variable is because js is build before scaladoc,
+   * so we cannot access its args
+   */
+  def markdownMain(): Unit =
```

Comment:
Why do we need two “main”s?

Reply:
Built scala.js output would run all modules. We don't want them to be run at docs.scala-lang, so we have to create the designated main for each target (scaladoc/docs.scala-lang). I was thinking about compile-time macro, but I think that we cannot provide env variable as the condition of the compile-time `inline match`